js/ui/date_box/ui.time_view.js

```diff
@@ -170,130 +176,149 @@ var TimeView = Editor.inherit({
+        editor.setAria("label", "minutes");
+        this._attachKeyboardProcessorToEditor(editor);
+
+        this[WIDGET_PART_NAMES.MINUTE] = editor;
     },
 
     _createFormat12Box: function() {
-        var periodNames = dateLocalization.getPeriodNames();
-        this._format12 = this._createComponent($("<div>").addClass(TIMEVIEW_FORMAT12_CLASS), SelectBox, extend({
+        const periodNames = getPeriodNames();
+        const editor = this._createComponent($("<div>").addClass(TIMEVIEW_FORMAT12_CLASS), SelectBox, {
             items: [
                 { value: TIMEVIEW_FORMAT12_AM, text: periodNames[0] },
                 { value: TIMEVIEW_FORMAT12_PM, text: periodNames[1] }
             ],
             valueExpr: "value",
             displayExpr: "text",
-            onValueChanged: (function(args) {
-                var hours = this._getValue().getHours(),
-                    time = new Date(this._getValue()),
-                    newHours = (hours + args.value * 12) % 24;
+            onValueChanged: ({ value }) => {
+                const hours = this._getValue().getHours();
+                const time = new Date(this._getValue());
+                const newHours = (hours + value * 12) % 24;
 
                 time.setHours(newHours);
                 this.option("value", time);
-            }).bind(this),
-            value: this._getValue().getHours() >= 12 ? TIMEVIEW_FORMAT12_PM : TIMEVIEW_FORMAT12_AM
-        }, {
+            },
+            value: this._getValue().getHours() >= 12 ? TIMEVIEW_FORMAT12_PM : TIMEVIEW_FORMAT12_AM,
             stylingMode: this.option("stylingMode")
-        }));
+        });
+
+        this._attachKeyboardProcessorToEditor(editor);
+        editor.setAria("label", "type");
 
-        this._format12.setAria("label", "type");
+        this[WIDGET_PART_NAMES.PERIOD_MARKER] = editor;
     },
 
     _refreshFormat12: function() {
         if(this.option("use24HourFormat")) return;
 
-        var value = this._getValue(),
-            hours = value.getHours(),
-            isPM = hours >= 12;
+        const value = this._getValue();
+        const hours = value.getHours();
+        const isPM = hours >= 12;
+        const newValue = isPM ? TIMEVIEW_FORMAT12_PM : TIMEVIEW_FORMAT12_AM;
+
+        this._silentEditorValueUpdate(this[WIDGET_PART_NAMES.PERIOD_MARKER], newValue);
+    },
 
-        this._format12._valueChangeActionSuppressed = true;
-        this._format12.option("value", isPM ? TIMEVIEW_FORMAT12_PM : TIMEVIEW_FORMAT12_AM);
-        this._format12._valueChangeActionSuppressed = false;
+    _silentEditorValueUpdate: function(editor, value) {
+        if(editor) {
+            editor._suppressValueChangeAction();
+            editor.option("value", value);
+            editor._resumeValueChangeAction();
+        }
     },
 
     _getNumberBoxConfig: function() {
-        return extend({
+        return {
             showSpinButtons: true,
-            disabled: this.option("disabled"),
```

Comment:
Why do we need this?

Reply:
When we create a nested widget via `createComponent` method, we say that root component should sync state of the `rtlEnabled` and `disabled` option with the nested one. So there is no need to set the initial value for this option

Reply:
Ok. Thank you!